drivers/netdev_ieee802154/netdev_ieee802154.c

```diff
@@ -152,7 +164,14 @@ int netdev_ieee802154_get(netdev_ieee802154_t *dev, netopt_t opt, void *value,
 #endif
         case NETOPT_MAX_PDU_SIZE:
             assert(max_len >= sizeof(int16_t));
-            *((uint16_t *)value) = (IEEE802154_FRAME_LEN_MAX -
+
+            /* first determine the PHY mode */
+            res = dev->netdev.driver->get(&dev->netdev, NETOPT_IEEE802154_PHY, value, sizeof(uint8_t));
+            if (res < 0) {
+                *(uint8_t*)value = 0;
```

Comment:
Not sure it is such a good idea, to change `value` in an error case. Better use a new variable for that.